src/java.base/linux/native/libsimdsort/avx2-linux-qsort.cpp

```diff
@@ -39,15 +37,9 @@ extern "C" {
             case JVM_T_INT:
                 avx2_fast_sort((int32_t*)array, from_index, to_index, INSERTION_SORT_THRESHOLD_32BIT);
                 break;
-            case JVM_T_LONG:
-                avx2_fast_sort((int64_t*)array, from_index, to_index, INSERTION_SORT_THRESHOLD_64BIT);
-                break;
             case JVM_T_FLOAT:
                 avx2_fast_sort((float*)array, from_index, to_index, INSERTION_SORT_THRESHOLD_32BIT);
                 break;
-            case JVM_T_DOUBLE:
-                avx2_fast_sort((double*)array, from_index, to_index, INSERTION_SORT_THRESHOLD_64BIT);
-                break;
```

Comment:
Please add safe assertions for missing types.

Reply:
This is from an older (but outdated) commit. The assertions have been added in other cases.